### PR TITLE
Add visualizer for caliban

### PR DESCRIPTION
**Pull request by tddough98**
Similar to #317, this introduces a new build of DeepCell Label for visualizing results created by the caliban model on deepcell.org/predict. I also work through the changes

There are also changes to work towards editing lineage labels. Editing the tracks is not fully functional at the moment. However, as this does
I plan on addressing this together with overlapping labels, as editing overlapping labels and editing the lineage requires sending back multiple files in a ZIP. This approach is already mocked up in the overlapping labels branch, so by merging it in, I'll be able to extend it to included updated lineages as well.

The core pieces of work remaining before the full editing app can be used in production are
- client side undo/redo
- client side persistence of changes
- 

The changes core to visualizing caliban output include

- Load and write lineage.json to the project archive
- Load lineage.json on the client 
- Add lineageMachine to manage the lineage
- Replace useTracking hook with useLineage and useEditLineage
  - Accessing the lineage happens through useLineage
  - Editing the lineage happens through useEditLineage
- Manage which cell is selected with lineageMachine
- Introduce a useEditing hook that returns whether the app is a visualizer or the full editing build
- Refactor components defined in Timeline.js into separate files to make them easier to work with and reuse

Some other changes included here are

- Fix issue with proxying the flask server while developing by adding setupProxy.js instead of the proxy field in package.json
 - Improve adding mousetrap class to input elements with `useMousetrapRef` hook
 - Parameterize which S3 bucket project files are uploaded to with the S3_BUCKET environment variable
 - Remove an extra copy of FrameSlider.js
 
This build is hosted at [tracks.deepcell.org](https://tracks.deepcell.org). You submit caliban output to be visualized with the same form and API route as the spots visualizer
```
curl --location --request POST 'tracks.deepcell.org/api/project' \
--form 'images="URL_TO_DOWNLOAD_CALIBAN_INPUT"' \
--form 'labels="URL_TO_DOWNLOAD_CALIBAN_OUTPUT"'
```

### CODE REVIEW

**Comment by msschwartz21** [comment]
I want to make a note that the visualizer on the data format that is output by the kiosk caliban consumer. The raw data is stored as a tif stack in one zip file and then the lineage.json and labeled data tif stack are stored in another zip file.

File: visualizer/src/Project/Load.js [hunk 1]
```diff
@@ -6,22 +6,25 @@ import ProjectContext from './ProjectContext';
 import createLoadMachine from './service/loadMachine';
 import createProjectMachine from './service/projectMachine';
 
-function Load({ id, track, spots }) {
+function Load({ id, spots }) {
```
Comment (msschwartz21) [inline visualizer/src/Project/Load.js:9]: Potentially a naive question, but why is spots still getting passed in here but track is not?

**Review by msschwartz21 — APPROVED** [review]
The visualizer is working really well and is a great addition to DCL functionality! 

How hard would it be to add back the ability to load trk files? Currently in order to view files that we already have saved as trks the data has to be resaved to the new zip format before uploading to the bucket. It would be convenient to be able to directly ingest trk files. However given that the visualizer's primary purpose is to connect to the kiosk, I'm ok with leaving out the trk functionality. Let me know what you think.

File: deepcell_label/cells.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+"""Computes label metadata dictionaries from label arrays."""
```
[the remark this answers is not in the document]
Comment (tddough98) [reply deepcell_label/cells.py:1]: Kinda... it's a more of a holdover version of cell_info from past versions. There are still a couple of components that use it to know what cells are in the project, namely labelsMachine to make a colormap with the correct number of colors and selectMachine to know what labels to cycle through.

File: deepcell_label/loaders.py [hunk 3]
```diff
@@ -166,6 +174,27 @@ def load_spots(f):
...
 
+def load_lineage(f):
+    """
+    Load lineage data from label file.
+
+    Args:
+        zf: file with zipped csv containing spots data
+
+    Returns:
+        bytes read from csv in zip or None if no csv in zip
```
[the remark this answers is not in the document]
Comment (tddough98) [reply deepcell_label/loaders.py:185]: Caught this in a later PR, so I moved the updated docstring here too

File: visualizer/src/Load/FileUpload.js [hunk 4]
```diff
@@ -97,7 +97,7 @@ export default function FileUpload({ loadService }) {
                   color='error'
                   key={file.path}
                 >
-                  {file.path} is not a valid file. Please upload a .tiff, .png, or .npz.
+                  {file.path} is not a valid file. Please upload a .zip, .tiff, or .png.
```
[the remark this answers is not in the document]
Comment (tddough98) [reply visualizer/src/Load/FileUpload.js:100]: I will update this to say 

> Please upload an image file (tiff, png, or npz) or a zip with both images and labels (tiff or npz for segmentations, json for lineage, or csv for spots)

File: visualizer/src/Project/Load.js (hunk 1 above)
Comment (tddough98) [reply visualizer/src/Project/Load.js:9]: Good catch, spots is not actually passed to the component and should be removed here

**Review by msschwartz21 — APPROVED** [review]

**Pull request merged**